### ISSUE TITLE
RJ45 with ESXi

### ISSUE THREAD

**Issue by fdabrao** [coreemu/core#187]
Hello all,

Is there possible CORE working with esxi ? I tried with simple router connected to RJ45 trying to ping external ip from real network in the same subnet, but there is no response.

![image](https://user-images.githubusercontent.com/17049832/45463939-8da50100-b6e4-11e8-901d-a7d88239f5b5.png)

Regards,

Fernando

**Comment by claudiudanilov** [comment]
I’ve seen that before, for me my ARP reply messages were being blocked on the VM interface. You can check that if you run tcpdump both on n1 inside CORE and on a machine on the same subnet in your real network. If that’s the case, what fixed it for me was setting up the VM interface to promiscuous mode.

Claudiu

**Comment by kd8bny** [comment]
@claudiudanilov Thanks! this fixed my headache

**Comment by coder-shanks** [comment]
@claudiudanilov Even though I set the interface in promiscuous mode, I was not able to ping the interface from router connected to it. I am using Wifi adapter as an interface to RJ45. Does it have to be an Ethernet interface to work?

**Comment by claudiudanilov** [comment]
If you tcpdump on the interface (both in and out of CORE) do you see any ARP messages going back and forth? I remember that without setting the interface in promiscuous mode we  were seeing broadcast ARPs only on one side of the interface, and then enabling promiscuous allowed those ARPs to go through. If you don’t see ARP messages going through then there maybe something else. In my environment I only tried it with wired interfaces but not sure WiFi would be any different.

Claudiu

**Comment by coder-shanks** [comment]
I see ARP messages going through only router interface and not through RJ45. So what would be the catch here?

**Comment by claudiudanilov** [comment]
I’d check to see whether your VM can ping another host through that interface, without CORE running. If that doesn’t work, likely you’ll need to set up your VM up properly first. If it does work, then there must be another issue that I’m not familiar with.

Claudiu

**Comment by coder-shanks** [comment]
Oops..sorry I should have told you earlier. I have set up the CORE on standalone ubuntu machine and Wifi adapter is connected to it. What I am trying to do here is to connect Wifi adapter interface IP through RJ45 in CORE. My end goal is to introduce the virtual traffic in the host machine and study its effect. Is it possible? Could you give me directions on that one? Your help is much appreciated. I am trying to get my head around it for a long time now.

**Comment by claudiudanilov** [comment]
The promiscuous mode fix I mentioned here originally was in the context of running CORE within esxi. It does not apply if you don’t use a VM environment.

What you describe should work out of the box, in general, though your specific WiFi adapter may not be fully supported. Try using a regular Ethernet port and see if that changes anything.

Claudiu

**Comment by coder-shanks** [comment]
Not sure. What is the problem with wifi adapter, but Ethernet port worked fine. I was able to see some ARP messages in Tcpdump. Can't we ping the host IP from any of the virtual nodes in CORE?

**Comment by bharnden** [comment]
No you cannot, the nodes are in a separate network namespace.

**Comment by coder-shanks** [comment]
So how will we come to know the effects of delay and loss configured in CORE on the host machine? Any suggestions.

**Comment by bharnden** [comment]
You typically would be running tools related to sending data within a CORE node itself. Double click a node to bring up a node's shell.

Then leveraging something like [mgen](https://www.nrl.navy.mil/itd/ncs/products/mgen) that would allow you to measure statistics about what you are doing. You could do anything that would make sense at this point.

**Comment by coder-shanks** [comment]
@bharnden @fdabrao @claudiudanilov I can now connect to physical device via RJ45 and able to send traffic on it. If say, I am running an application on my host PC which connects with the same physical device. Does the traffic running from CORE have any effect on the network performance of that application? If yes, how can we achieve that? Please refer the below diagram for reference. 

![image](https://user-images.githubusercontent.com/36691389/55064791-98335080-50a0-11e9-9d8f-d45054f5dc8e.png)

**Comment by coder-shanks** [comment]
Do you have any comments on the above? It would be really helpful if you provide ANY valuable suggestions. @fdabrao @claudiudanilov @kd8bny @bharnden

**Comment by coder-shanks** [comment]
> Do you have any comments on the above? It would be really helpful if you provide ANY valuable suggestions. @fdabrao @claudiudanilov @kd8bny @bharnden

Is this even possible using CORE emulator? Otherwise I will just be wasting time here :(

**Comment by bharnden** [comment]
Please open a new issue with a clear and complete use case if help is still needed, hope you figured it out.
